pkg/util/fxutil/fxoptional/component.go

```diff
@@ -0,0 +1,98 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fxoptional
+
+import "github.com/DataDog/datadog-agent/pkg/util/optional"
+
+// Component is an optional component type.
+// The main purpose of this component is to make it explicit that the component may not be present.
+// Here is an full example of how to use it:
+//
+//	// ------------------------------------------
+//	// Component.go
+//	// ------------------------------------------
+//	type Component fxoptional.Component[MyInterface]
```

Comment:
I find myself constantly confused by using the same name to represent two different things, especially when those things are very similar except in one very important way. In my own head I have trouble keeping the terms straight and think this might remain a problem going forward. Even in this review discussion, folks have to differentiate between `Component`, `fxoptional.Component`, and `the component` to mean the wrapped object. I would much rather have a different term for this, such as `fxoptional.Maybe` or `fxoptional.Retriever` or `fxoptional.Creator` to represent the interface that might or might not return an actual Component.